File: java/Easy/FindIndexOfFirstOccurrenceInString.java
package Easy;

public class FindIndexOfFirstOccurrenceInString {
    public static void main(String[] args){
        String a = "ssadbutsad";
        String b = "sad";
        System.out.println(strStr(a, b));
    }

    public static int strStr(String haystack, String needle){
        for(int i = 0; i <= haystack.length() - needle.length(); i++){
            outerloop:
            if(needle.charAt(0) == haystack.charAt(i)){
                for(int j = 1; j < needle.length(); j++){
                    if(haystack.charAt(i + j) != needle.charAt(j)) break outerloop;
                }
                return i;
            }
            
        }
        return -1;
    }
}

File: kotlin/Easy/FindIndexOfFirstOccurrenceInString.kt
package Easy

class FindIndexOfFirstOccurrenceInString{
    fun strStr(haystack: String, needle: String): Int {
        for(i in 0 .. haystack.length - needle.length){
            label@
            while(haystack[i] == needle[0]){
                for(j in 1 until needle.length){
                    if(haystack[i + j] != needle[j]){
                        break@label
                    }
                }
                return i
            }
        }
        return -1
    }
}

fun main () {

    var a = FindIndexOfFirstOccurrenceInString().strStr("adaadfsd", "adf")
    println(a)
}